packages/flutter_tools/lib/src/runner/flutter_command_runner.dart

```diff
@@ -71,6 +71,11 @@ class FlutterCommandRunner extends CommandRunner<Null> {
         negatable: true,
         hide: !verboseHelp,
         help: 'Whether to use terminal colors.');
+    argParser.addFlag('version-check',
+        negatable: true,
+        defaultsTo: true,
+        hide: !verboseHelp,
+        help: 'Allow Flutter to check for updates when this command runs.');
```

Comment:
`Check for updates to Flutter when running a command.`?

Reply:
I wasn't sure if that would be misleading, since this doesn't actually cause an update check, it just makes the original warning not appear if there's no new version. It's possible you can pass `--update-check` and it still won't tell you there's an update for 30 days(?). We could of course probably reduce that period now we have this change?

Reply:
That said, we might want to review the logic a little if we do reduce that period, to ensure if someone has an unreliable update check, they're not suddenly seeing the warning way more often. Probably we need two dates - one for how old it needs to be for us to do the check, and another for how long before we warn the user their version is old if we couldn't verify there was a new one. The first should probably be < 7 days but the second probably still wants to be > 30 days. Possibly that all comes under #14920.

Reply:
@devoncarew Landing this as-is for now, but happy to tweak text if you have an opinion on the above.